jigsaw-game/client/menu.py

```diff
@@ -0,0 +1,110 @@
+import pygame
+from pygame.locals import QUIT, MOUSEBUTTONDOWN
+
+pygame.init()
+
+# Colors
+WHITE = (255, 255, 255)
+RED = (255, 0, 0)
+BLUE = (0, 0, 255)
+
+# Screen dimensions
+WIDTH, HEIGHT = 800, 600
+screen = pygame.display.set_mode((WIDTH, HEIGHT))
+pygame.display.set_caption('Game Menu')
+
+font = pygame.font.SysFont(None, 36)  # Default font, size 36
+
+
+def draw_button(surface, rect, text, button_color, text_color):
+    pygame.draw.rect(surface, button_color, rect)
+    txt = font.render(text, True, text_color)
+    txt_pos = (rect.x + rect.width // 2 - txt.get_width() // 2, rect.y + rect.height // 2 - txt.get_height() // 2)
+    surface.blit(txt, txt_pos)
+
+
+def is_button_clicked(event, rect):
+    if event.type == MOUSEBUTTONDOWN:
+        return rect.collidepoint(event.pos)
+    return False
+
+
+# Views
+def game_selection_view():
+    screen.fill(WHITE)
+
+    easy_button = pygame.Rect(100, 100, 200, 50)
+    moderate_button = pygame.Rect(100, 200, 200, 50)
+    hard_button = pygame.Rect(100, 300, 200, 50)
```

Comment:
for **Quick Game** Easy, Intermediate and Hard are good options, but for **Custom Game** we should include either text entry or better still a slider for how many "tiles"... but make it follow squared numbers 4, 9, 16, 25, 36, 49, 64, 81, etc.